linux/root_test.go

```diff
@@ -0,0 +1,87 @@
+// +build linux,!skipIntegration
+
+// nolint:errcheck,gomnd
+package linux_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/anuvu/disko"
+	"github.com/anuvu/disko/linux"
+	"github.com/anuvu/disko/partid"
+	"github.com/stretchr/testify/assert"
+)
+
+const MiB = 1024 * 1024
+const GiB = MiB * 1024
+
+// runLog - run command and Printf, useful for debugging errors.
+func runLog(args ...string) {
+	out, err, rc := runCommandWithOutputErrorRc(args...)
+	fmt.Printf("%s\n", cmdString(args, out, err, rc))
+}
+
+func TestPartition(t *testing.T) {
+	skipIfNoLoop(t)
+	var loopDev string
+
+	ast := assert.New(t)
+	tmpFile := getTempFile(GiB)
+
+	defer os.Remove(tmpFile)
+
+	if cleanup, path, err := connectLoop(tmpFile); err != nil {
+		runLog("losetup", "-a")
+		t.Fatalf("failed loop: %s\n", err)
+	} else {
+		defer cleanup()
+		loopDev = path
+	}
+
+	lSys := linux.System()
+
+	disk, err := lSys.ScanDisk(loopDev)
+	if err != nil {
+		t.Fatalf("Failed first scan of %s: %s\n", loopDev, err)
+	}
+
+	part1 := disko.Partition{
+		Start:  disk.FreeSpaces()[0].Start,
+		Last:   disk.FreeSpaces()[0].Start + (100 * MiB) - 1,
+		Type:   disko.PartType(partid.LinuxRootX86),
+		Name:   randStr(10),
+		Number: 1,
+	}
+
+	// part3 leaves 100MiB gap to verify FreeSpaces
+	part3 := disko.Partition{
+		Start:  part1.Last + 100*MiB + 1,
+		Last:   disk.FreeSpaces()[0].Last,
+		Type:   disko.PartType(partid.LinuxFS),
+		Name:   randStr(8),
+		Number: 3,
+	}
+
+	if err = lSys.CreatePartition(disk, part1); err != nil {
```

Comment:
Not that you need to fix it, but since you're already depending on testify anyway, you can shorten stuff like this with `assert.NoError(...)`

Reply:
oh. yeah, i can do that. i just kind of threw testify in at the end.